assets/blocks/view-results-block/index.js

```diff
@@ -38,4 +42,21 @@ export default createButtonBlockType( {
 		),
 		validPostTypes: [ 'course' ],
 	},
+	EditWrapper: ( { children } ) => {
```

Comment:
Since we use it 3 times, I wonder if it would make sense to extract it to an external function. WDYT?

Reply:
I have mixed feelings about this. I don't feel good about extracting the whole function, because that feels inflexible. What if one of the blocks needs to make a tweak to how the `edit` component works, and it needs to be different from the other blocks?

What I did in c059a1851 is moved the context handling to inside the toolbar component, so that logic doesn't need to be repeated.

I was debating whether to use an attribute (as in this case) or to just create a new wrapper component (e.g. `CourseStatusToolbarWithStatusFromContext`). Using the attribute seemed simpler, but it makes the Toolbar component a little more complex. Maybe a wrapper component would be nicer?

Reply:
> What I did in https://github.com/Automattic/sensei/commit/c059a1851f8af02ca662a40219d8bb5ac072ef46 is moved the context handling to inside the toolbar component, so that logic doesn't need to be repeated.

Looks good to me! I think your idea on keeping the props there is to make this component more open and flexible, right?

> I was debating whether to use an attribute (as in this case) or to just create a new wrapper component (e.g. CourseStatusToolbarWithStatusFromContext). Using the attribute seemed simpler, but it makes the Toolbar component a little more complex. Maybe a wrapper component would be nicer?

A pattern that comes to my mind for this case would be "compound components", where we could export the provider "tied" to the component that we export the toolbar (making it explicitly dependent). But the prop, as you implemented it, looks good to me too.

Reply:
> I think your idea on keeping the props there is to make this component more open and flexible, right?

Do you mean the `courseStatus` and `setCourseStatus` props? They are needed by the `CourseActionsEdit` component itself, since it uses those variables directly, rather than getting them from a context - https://github.com/Automattic/sensei/pull/6324/files#diff-10a47f6512f92a19da345493f87b244e6e3e645d54f4aa8f992eba894fadccb6R147

I suppose another option might have been to _always_ use the context within `CourseStatusToolbar`, and just add another `CourseStatusContext.Provider` wrapper around the toolbar render.

But I do kinda like having the props that can be used directly 🤷‍♂️ 

> A pattern that comes to my mind for this case would be "compound components", where we could export the provider "tied" to the component that we export the toolbar (making it explicitly dependent). But the prop, as you implemented it, looks good to me too.

👍

Reply:
> Do you mean the `courseStatus` and `setCourseStatus` props? They are needed by the CourseActionsEdit component itself

Yep! I missed that one! :)